### DISCUSSION TITLE
Popup when Spirit chooses the question

### DISCUSSION THREAD

**Discussion by Ramenhotep** [Oliver-Akins/Phantom-Ink-Online#84] [Ideas]
Not a feature I'm requesting, but I had a UX idea: I think it would work well for the medium(s) of a team to get a banner popup (like the error banner) with the question text when their Spirit chooses the question. That way they don't need to open the Past Questions to check which has been a bit of a hard thing to explain.

**Comment by Eldritch-Oliver** [comment]
This is definitely possible, though it would require a bit of modifications to the client and server. Because right now the game only checks for questions every 5 seconds when the questions insert is visible (if you love to insert open it keeps pinging the server), beyond that the client doesn't actually keep track of the questions, meaning that this would require a special event being emitted from the server in order get the client to properly acknowledge it. None of this would be hard to do, but it would be a few hours to implement because of the changes that are needed.

**Comment by Eldritch-Oliver** [comment]
Another possible solution is that we could decrease the amount of time between each server poll, because right now it's at five seconds, we could probably drop it to 3 seconds without too much additional server strain.

Or, we could add a little indicator that it will poll the server in X seconds (IE: "Refreshing in X seconds" where X is the number of seconds left until it get's the questions)


Both of these would require less restructuring than the previously commented solution. The previous solution would take ~6 hours to implement and test in the worst case, ~2 in the best case. Reducing the timer delay would be less than 5 minutes, the countdown would take slightly longer at ~1 hour best case, ~2 hours worst case. These times are just going based off the sheer scale of the codebase that would need to be modified to account for them.

Roughly what would need to changed in the code:
- Alert bar similar to error message: Server's `SendCard` event listener, client's either App.vue, Hand.vue, PreviousQuestions.vue, or GameBoard.vue (having to bubble event up in any non-App.vue case)
- Reduce time (no countdown): a single number in PreviousQuestions.vue
- Timer (with or without time reduction): the interval in PreviousQuestions.vue and a little bit of styling to position it in a decent spot.

**Comment by Ramenhotep** [comment]
So the problem is that the client doesn't actually get the questions until the insert is opened? Ah, well then it seems like there isn't a good solution.

**Comment by Eldritch-Oliver** [comment]
Yeah, the client only gets the questions when you press the show past questions button and every 5 seconds after that.
